pkg/session/session.go

```diff
@@ -133,7 +134,18 @@ func GetOrCreate(ctx context.Context, params *Params) (*Session, error) {
 
 		vimSessionActive, err := s.SessionManager.SessionIsActive(ctx)
```

Comment:
Did you consider using "UserSession" as mentioned in the issue request, we have hit the same issue in our code and moved to the below piece of code for checking the active session and it is working well for us. Do you see any issues with the approach?

u, err := s.SessionManager.UserSession(ctx)
if err != nil {
	if soap.IsSoapFault(err) {
		fault := soap.ToSoapFault(err).VimFault()
		// If the PropertyCollector is not found, the saved session for this URL is not valid
		if _, ok := fault.(vimtypes.ManagedObjectNotFound); ok {
			return false, nil
		}
	}
	return false, err
}
On a datacenter scoped vCenter account, SessionManager.SessionIsActive() fails with permission denied error.
Need to change it to SessionManager.UserSession() method, which does not require special permissions.

Reply:
Thanks.
I validated in an environment and indeed the `UserSession` would be a better choice as it does not require any permission at all.